README.md

```diff
@@ -27,58 +26,70 @@ Repository to discuss the new REST API contract for DSpace 7. The code to implem
+
+- `GET`:
+Returns the first page of the resources in the group (or collection). See [Pagination](#pagination) section. 
 
 ### On single resource endpoints
-- GET
-Returns a single entity.
 
-- HEAD
-Returns whether the item resource is available.
+This type of endpoint interacts with a single resource (object). For example: `/api/core/items/123-456-789` references a single Item.
+
+- `GET`:
+Returns a single resource.
+
+- `HEAD`:
+Returns whether the target resource is available.
 
-- PUT
-Replaces the state of the target resource with the supplied request body.
+- `PUT`:
+Replaces the state of the target resource with the supplied request body. This updates the object (via full replacement). The updated information _should be included_ in the request body. Related or additional information (such as required associations to other objects) may be passed as querystring parameters in the request, _if it is necessary to update the object._
 
-- PATCH
+- `PATCH`:
 Similar to PUT but partially updating the resources state. We adhere to the [JSON Patch specification RFC6902](https://tools.ietf.org/html/rfc6902) see the [General rules for the Patch operation](patch.md) for more details.
 
-- DELETE
-Deletes the resource exposed.
+- `DELETE`:
+Deletes the target resource and object.
 
-### On sub-path of a single resource endpoint)
-- GET
-Returns the state of the association resource
+### On sub-path of a single resource endpoint (associations)
 
-- PUT
-Binds the resource pointed to by the given URI(s) to the resource. Return 400 Bad Request if multiple URIs were given for a to-one-association
+This type of endpoint interacts with (one or more) resources that are *associated with* a single resource. For that reason, it is sometimes called an "association" endpoint. For example: `/api/core/items/123-456-789/mappedCollections` references all Collections that are mapped to a single Item.
 
-- POST
-Only supported for collection associations. Adds a new element to the collection.
+- `GET`:
+Returns the state of the association (for the current resource)
 
-- DELETE
-Unbinds the association. Return 405 Method Not Allowed if the association is non-optional
+- `PUT`:
+Binds (or links) the resource(s) pointed to by the given URI(s) to the current resource. Resource URIs to link must sent in the body of the request using `Content-Type:text/uri-list`. [See example from Spring Data REST Relationships](https://www.baeldung.com/spring-data-rest-relationships). Return `400 Bad Request` if multiple URIs were given for a *-to-one-association
+
+- `POST`:
+Only supported for collection associations (i.e. associations allowing for multiple resources). Adds/Links a new resource (via its URI) to the association. Resource URIs to link must sent in the body of the request using `Content-Type:text/uri-list`. [See example from Spring Data REST Relationships](https://www.baeldung.com/spring-data-rest-relationships)
```

Comment:
TODO for self. This also needs a more specific example of what a POST would look like using "text-uri-list".